src/misc/msccl/msccl_setup.cc

```diff
@@ -422,7 +426,8 @@ ncclResult_t mscclSetupKernel(const void* sendBuff, void* recvBuff, size_t count
 
   mscclWork work;
   work.syncFlags = status.syncFlags;
-  work.scratchBuffer = status.scratchBuffer;
+  size_t sizeNeeded = computeSizeNeeded(status.nBytes, hostAlgo->nScratchChunks, hostAlgo->nChunksPerLoop);
+  work.scratchBuffer = status.scratchBuffers[sizeNeeded];
```

Comment:
maybe move the code inside mscclSetupScratch to here? Is there a need for 2-steps?

Reply:
submitted a new commit.